core/src/androidTest/java/org/hisp/dhis/android/core/category/CategoryOptionComboCategoryLinkStoreShould.java

```diff
@@ -33,55 +41,86 @@ public void setUp() throws IOException {
+        thenAssertNewCategoryOptionComboLinkWasInserted();
+
+        whenDeleteACategoryOptionComboCategoryLink();
+
+        thenAssertThatThereAreNotCategoryComboCategoryLinkInDB();
+    }
+
+    @Test
+    @MediumTest
+    public void delete_all_category_option_combo_category_link_from_db() {
+        whenInsertNewCategoryOptionComboCategoryLink();
+
+        thenAssertNewCategoryOptionComboLinkWasInserted();
+
+        whenDeleteAllCategoryOptionComboCategoryLinks();
+
+        thenAssertThatThereAreNotCategoryComboCategoryLinkInDB();
+    }
+
+    @Test
+    @MediumTest
+    public void update_a_category_option_combo_category_link() {
+        whenInsertNewCategoryOptionComboCategoryLink();
+
+        thenAssertNewCategoryOptionComboLinkWasInserted();
+
+        whenUpdateANewCategoryOptionComboCategoryLink();
+
+        thenAssertThatNewCategoryOptionComboCategoryLinkWasUpdated();
+
+    }
+
+    @Test
+    @MediumTest
+    public void query_all_category_option_combo_category_link() {
+        whenInsertNewCategoryOptionComboCategoryLink();
+
+        thenAssertNewCategoryOptionComboLinkWasInserted();
+
+        thenAssertThatQueryAllBringCategoryComboCategoryLinksInDB();
     }
 
     private void givenACategoryOption() {
-        Date today = new Date();
-
-        newCategoryOption = CategoryOption.builder()
-                .uid("TXGfLxZlInA")
-                .code("SECHN")
-                .created(today)
-                .name("SECHN")
-                .shortName("SECHN")
-                .displayName("SECHN")
-                .build();
+        newCategoryOption = generatedCategoryOption(DEFAULT_CATEGORY_OPTION_UID);
     }
 
     private void givenACategoryOptionCombo() {
-        Date today = new Date();
-
-        newCategoryOptionCombo = CategoryOptionCombo.builder()
-                .uid("NZAKyj67WW2")
-                .created(today)
-                .name("SECHN, Male")
-                .shortName("SECHN, Male")
-                .displayName("SECHN, Male")
-                .build();
+        newCategoryOptionCombo = generateCategoryOptionCombo(DEFAULT_CATEGORY_OPTION_COMBO_UID);
     }
 
     private void givenACategoryOptionComboCategoryLink() {
         newCategoryOptionComboCategoryLink = CategoryOptionComboCategoryLinkModel.builder()
-                .optionCombo("NZAKyj67WW2")
-                .category("TXGfLxZlInA")
+                .category(DEFAULT_CATEGORY_OPTION_UID)
+                .categoryOptionCombo(DEFAULT_CATEGORY_OPTION_COMBO_UID)
                 .build();
     }
 
-    private void whenInsertNewCategoryOption() {
+    private void whenInsertNewCategoryOption(CategoryOption categoryOption) {
         CategoryOptionStoreImpl store = new CategoryOptionStoreImpl(databaseAdapter());
-        store.insert(newCategoryOption);
+        store.insert(categoryOption);
     }
 
     private void whenInsertNewCategoryOptionComboCategoryLink() {
+        givenACategoryOption();
```

Comment:
this method has two responsibilities, create a fixture (givens...) and realize actions (whens..), I think that the code would be more readable in tests where is used if split this method in two, one responsibility per method, can you refactor it?

Reply:
Sure!